broker-daemon/state-machines/fill-state-machine.js

```diff
@@ -111,10 +114,11 @@ const FillStateMachine = StateMachine.factory({
    * @param  {RelayerClient}       options.relayer
    * @param  {Map<String, Engine>} options.engines     Collection of all avialable engines
    * @param  {Function}            options.onRejection A function to handle rejections of the fill
+   * @param  {Function}            options.onCompletion A function to handle the completion of the fill
    * @return {Object}                                  Data to attach to the state machine
    */
-  data: function ({ store, logger, relayer, engines, onRejection = function () {} }) {
-    return { store, logger, relayer, engines, onRejection, fill: {} }
+  data: function ({ store, logger, relayer, engines, worker }) {
```

Comment:
I'm not a fan of exposing the block order worker to the state machine. I think the worker should know about the state machine, but not the inverse of that. I don't think there is a need to make that relationship two-way.

Reply:
@treygriffith its not necessarily exposing the worker as much as exposing the events API. The current use of `onRejection` was ok, but its real goal is message passing. The only reason we use `worker` here is for triggering an event that the block order is listening for, that the block order can only set.

Reply:
@treygriffith This was the tradeoff that I made vs using a callback. I tried to avoid this situation because we both had agreed earlier that we wanted the worker/state machine separation. This was one of the cleanest implementation choices